scripts/test

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+# This script runs cargo test with the most common testing configurations.
+# Arguments given will be passed through to "cargo test"
+# This runs in the Makefile environment via "make run"
+
+set -euo pipefail
+
+# Run from the Makefile environment
+MAKEFILE_RUN=${MAKEFILE_RUN:-""}
+if [[ -z $MAKEFILE_RUN ]] ; then
+  COMMAND="$0 $*" exec make run
```

Comment:
So if I understand it correctly, `make test` will invoke `scripts/test`, which invokes `make run`, which invokes `scripts/test` again?

Reply:
yes, exactly

Reply:
oh, there is one thing wrong with that statement: `make test` will invoke `./scripts/test-all`. `./scripts/test` is a more basic wrapper for the way we run `cargo test`.